src/assert_output.bash

```diff
@@ -187,11 +192,17 @@ assert_output() {
     fi
   else
     if [[ $output != "$expected" ]]; then
-      batslib_print_kv_single_or_multi 8 \
-      'expected' "$expected" \
-      'actual'   "$output" \
-      | batslib_decorate 'output differs' \
-      | fail
+      if (( show_diff )); then
+        diff <(echo "$expected") <(echo "$output") \
+        | batslib_decorate 'output differs' \
+        | fail
+      else
+        batslib_print_kv_single_or_multi 8 \
```

Comment:
Perhaps batslib_print_kv_single_or_multi` could grow support for diffing expected and actual?

For instance it could use `wdiff` instead of `diff` if it detects single-line text.